### ISSUE TITLE
Content Contribution: Architecture -> The Serverless Architectural Pattern section

### ISSUE THREAD

**Issue by rupakg** [serverless/guide#52]
## Description

Add content to the `Architecture -> The Serverless Architectural Pattern` section of the guide.

**Filename:** `/guide/source/architecture/serverless_arch_pattern.md`

**Page/Sub-Section:** Architecture -> The Serverless Architectural Pattern

**Contributing Author:** Rupak Ganguly (Serverless Inc.) 

**Details:** 

#### Introduction
- Why do we need a new architectural pattern?
- Functions as a Service (FaaS)
- How do we get there?
   - Need for agile development lifecycles
   - Need for independent teams working in parallel
   - Focus on building than managing infrastructure
- SOA -> microservices -> functions (serverless)
- serverless is like microservices but more granular
- serverless architecture: FaaS for compute, BaaS (backend-as-a-service) for everything else (DB, search, cache, CDN)

#### How serverless architecture looks like?
- serverless architectures are opinionated microservices architectures
- different serverless arch:
    - greenfield - the whole system in functions
    - adding features to a legacy system with functions (communication via SDK faster, or API Gateway endpoint - latency issue)

#### Challenges
- cold start
- API Gateway latency
- invocation limit
- distributed tracing
- centralized logging

#### Migration 
- from traditional SOA architecture
- from microservices architecture
- grouping function is good (they can use the same DB)
- taking out functionalities from the existing system

Link to presentation when I cover some of the stuff from here: https://speakerdeck.com/mthenw/the-state-of-serverless-1

**Comment by rupakg** [comment]
Duplicate of #54
